src/main_WorldViewer.cpp

```diff
@@ -44,25 +62,26 @@ class REGothWorldViewer : public REGoth::REGothEngine
   {
     using namespace REGoth;
 
-    if (mWorld.empty())
+    if (mConfig.world.empty())
     {
       REGOTH_THROW(InvalidStateException, "World cannot be empty.");
     }
 
-    bs::StringUtil::toUpperCase(mWorld);
-    if (not bs::StringUtil::endsWith(mWorld, ".ZEN"))
+    bs::String worldName = mConfig.world;
+    bs::StringUtil::toUpperCase(worldName);
+    if (not bs::StringUtil::endsWith(worldName, ".ZEN"))
```

Comment:
Could we somehow put transformations like this into the config class so the game code doesn't have to deal with it?

Reply:
I've added a neat `verifyCLIOptions` method which will be called as last step after the parameter parsing, where the config object ca be checked for integrity (and throw if not) and inputs can be sanitised. I use this to make the engine path and the original assets path absolute now as well.